static/src/javascripts/projects/common/modules/discussion/comments.js

```diff
@@ -0,0 +1,588 @@
+
+        if (
+            showRepliesElem.length > 0 &&
+            !bonzo(showRepliesElem).hasClass('u-h')
+        ) {
+            showRepliesElem[0].click();
+        }
+
+        commentBox.render(parentCommentEl);
+
+        commentBox.on('post:success', (comment: commentType) => {
+            let responses = qwery('.d-thread--responses', parentCommentEl)[0];
+
+            if (!responses) {
+                responses = bonzo.create(
+                    '<ul class="d-thread d-thread--responses"></ul>'
+                )[0];
+                bonzo(parentCommentEl).append(responses);
+            }
+
+            this.destroy();
+            this.addComment(comment, responses);
+        });
+    }
+
+    reportComment(e: Event): void {
+        e.preventDefault();
+
+        const currentTarget: HTMLElement = (e.currentTarget: any);
+        const commentId = currentTarget.getAttribute('data-comment-id') || '';
+        const reportContainer = $(
+            `#comment-${commentId} .js-report-comment-container`
+        ).first();
+
+        const submitHandler = (form: HTMLFormElement) => {
+            form.removeAttribute('hidden');
+
+            bean.one(form, 'submit', (submitEvent: Event) => {
+                submitEvent.preventDefault();
+                const category: HTMLInputElement = (form.querySelector(
+                    '[name="category"]'
+                ): any);
+                const comment: HTMLInputElement = (form.querySelector(
+                    '[name="comment"]'
+                ): any);
+
+                if (category.value !== '0') {
+                    const email: HTMLInputElement = (form.querySelector(
+                        '[name="email"]'
+                    ): any);
+
+                    DiscussionApi.reportComment(commentId, {
+                        emailAddress: email.value,
+                        categoryId: category.value,
+                        reason: comment.value,
+                    })
+                        .then(() => this.reportCommentSuccess(form))
+                        .catch(() => this.reportCommentFailure());
+                }
+            });
+        };
+
+        $('.js-report-comment-form')
+            .first()
+            .each(submitHandler)
+            .appendTo(reportContainer);
+    }
+
+    // eslint-disable-next-line class-methods-use-this
+    reportCommentSuccess(form: HTMLElement): void {
+        form.setAttribute('hidden', '');
+    }
+
+    // eslint-disable-next-line class-methods-use-this
+    reportCommentFailure(): void {
+        const commentClose = document.querySelector('.d-report-comment__close');
+        const commentError = document.querySelector(
+            '.js-discussion__report-comment-error'
+        );
+
+        if (commentError) {
+            commentError.removeAttribute('hidden');
+        }
+
+        if (commentClose) {
+            commentClose.classList.add('d-report-comment__close--error');
+        }
+    }
```

Comment:
`reportCommentSuccess` and `reportCommentFailure` shouldn't be publicly-exposed methods, IMO. Their definition ought to be scoped to the `reportComment` method. Not sure it's worth refactoring at this point.